lightning/src/ln/msgs.rs

```diff
@@ -956,6 +956,33 @@ impl From<std::net::SocketAddr> for SocketAddress {
 		}
 }
 
+#[cfg(feature = "std")]
+impl std::net::ToSocketAddrs for SocketAddress {
+	type Iter = std::option::IntoIter<std::net::SocketAddr>;
+
+	fn to_socket_addrs(&self) -> std::io::Result<Self::Iter> {
+		match self {
+			SocketAddress::TcpIpV4 { addr, port } => {
+				let ip_addr = std::net::Ipv4Addr::from(*addr);
+				(ip_addr, *port).to_socket_addrs()
+			}
+			SocketAddress::TcpIpV6 { addr, port } => {
+				let ip_addr = std::net::Ipv6Addr::from(*addr);
+				(ip_addr, *port).to_socket_addrs()
+			}
+			SocketAddress::Hostname { ref hostname, port } => {
+				Ok((hostname.as_str(), *port).to_socket_addrs()?.next().into_iter())
+			}
+			SocketAddress::OnionV2(..) => {
+				Err(std::io::Error::from(std::io::ErrorKind::Unsupported))
```

Comment:
Ah looks like the MSRV (minimum supported Rust version) tests are failing because the `Unsupported` variant is only available from version 1.53 up. We'll need to use another variant here, probably `Other` until we bump the MSRV.